test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -130,6 +132,8 @@ public void tearDown() throws Exception {
         logger.trace("[{}#{}]: cleaning up after test", getTestClass().getSimpleName(), getTestName());
         awaitIndexShardCloseAsyncTasks();
         ensureNoInitializingShards();
+        ensureAllFreeContextActionsAreConsumed();
```

Comment:
Why can't we await for the tasks to disappear? We should be able to wait for the tasks to be completed.

Is this causing more tests to fail other than the single test case that this was originally flagged for?

I would think adding something like this to the sub-class tear down:
```
logger.info("--> waiting for all ongoing tasks to complete within a reasonable time");
        safeGet(clusterAdmin().prepareListTasks().setActions(FREE_CONTEXT_ACTION_NAME.name() + "*").setWaitForCompletion(true).execute());
```
would suffice.

Reply:
This seems to suffice indeed!  Thanks for the great alternative @benwtrent  !  Really wanted to avoid adding some much boilerplate code, but wasn't certain of the available feasible options for the mocked single node case, so I went with this. 

Will remove all other changes in favor of this one-line :D

Reply:
@pmpailis sorry to harp on this, but does this really need to be in `test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java` ? Why not only focus on the individual tests that have the issue? Are way more tests than just `SearchProgressActionListenerIT` failing?

Reply:
@pmpailis if no other tests than `SearchProgressActionListenerIT` are failing because of this, I would much prefer a `tearDown()` override, waiting there, and then calling `super.tearDown()`

Reply:
Unfortunately there were various tests failing for the same (seemingly) reason: 

* https://github.com/elastic/elasticsearch/issues/110409
* https://github.com/elastic/elasticsearch/issues/110227 ( and a few more  I think)

It seems that this is an issue for the whole suite of `ESSingleNodeTestCase` tests, hence why I opted for the class  `tearDown`  override.

Reply:
@pmpailis fabulous. I was not aware. Please make sure that this PR is mentioned in those various tests.